Sources/AWSLambdaRuntimeCore/LambdaRunner.swift

```diff
@@ -12,22 +12,31 @@
 //
 //===----------------------------------------------------------------------===//
 
+import AWSXRaySDK // TODO: use swift-tracing when available
+import Baggage
 import Dispatch
 import Logging
 import NIO
 
+// type names defined in `TracingInstrument`, aliases will be removed
+public typealias TracingInstrument = XRayRecorder
+public typealias NoOpTracingInstrument = XRayNoOpRecorder
+
 extension Lambda {
     /// LambdaRunner manages the Lambda runtime workflow, or business logic.
     internal final class Runner {
         private let runtimeClient: RuntimeClient
+        private let tracer: TracingInstrument
         private let eventLoop: EventLoop
         private let allocator: ByteBufferAllocator
 
         private var isGettingNextInvocation = false
 
-        init(eventLoop: EventLoop, configuration: Configuration) {
+        init(eventLoop: EventLoop, configuration: Configuration, tracer: TracingInstrument = NoOpTracingInstrument()) {
```

Comment:
since this internal I prefer no default and just pipe the right things explicitly